ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplicationMigrationTool.java

```diff
@@ -0,0 +1,537 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.exec.repl;
+
+import jodd.exception.UncheckedException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.ContentSummary;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.util.StringUtils;
+import org.apache.hadoop.util.Tool;
+import org.apache.hadoop.util.ToolRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+import static org.apache.hadoop.hdfs.protocol.OpenFilesIterator.OpenFilesType.ALL_OPEN_FILES;
+
+public class ReplicationMigrationTool implements Tool {
+
+  protected static transient Logger LOG = LoggerFactory.getLogger(ReplicationMigrationTool.class);
+  private Configuration conf;
+  private String help = "\tSample Usage: \n"
+      + "hive --replMigration -dumpFilePath <path to external table info file> [-dirLevelCheck] "
+      + "[-fileLevelCheck] [-verifyOpenFiles] [-verifyChecksum] [-filters] [-conf] [-queueSize] [-numThreads]"
+      + " [-checksumQueueSize] [-checksumNumThreads] [-help]\n"
+      + "-dumpFilePath: The fully qualified path to the external table info file. \n"
+      + "-dirLevelCheck: Validate at directory level"
+      + "-fileLevelCheck: Validate at file level"
+      + "-verifyOpenFiles: Validate there is no open files on the source path. \n"
+      + "-verifyChecksum: Whether the checksum needs to be validated for each file. Can not be used with "
```

Comment:
will fail for encry zones etc.

Reply:
Added, if the encryption zone is different or the checksum algo is different.